projects/components/src/multi-select/multi-select.component.ts

```diff
@@ -136,6 +136,9 @@ export class MultiSelectComponent<V> implements AfterContentInit, OnChanges {
   @Input()
   public enableSearch: boolean = false;
 
+  @Input()
+  public enableImplicitOptionsSearch: boolean = true;
```

Comment:
What if we took a page out of the table api (hah!) and took in some like a searchProvider object instead. We could default it to a client side version, but provide a server implementation in other places.

Reply:
oh god. let me check

Reply:
That sounds complicated. Can we do it with just may be a `searchMode` enum (Implicit and external)? I think eventually, we would have a separate basic filter multi select component, but we need this as a stepping stone.

Reply:
OK looked at multiselect - the search provider wouldn't work as well because the data is actually coming in as content children vs an array like some of the other components, so an output is the only way that comes to mind to accomplish this - but let's clean up the ambiguity between the two enable search inputs as you mentioned.

Reply:
so does this look good? 
```
@Input()
public enableSearch: boolean = false;

@Input()
public serachMode: MultiSelectSearchMode = MultiSelectSearchMode.Implicit;

...


export const enum MultiSelectSearchMode {
  Implicit = 'implicit,
  External = 'external
}
```

Reply:
I'd rather boil down another input and make the enum values more explicit, but same idea:
```
@Input()
public searchMode: MultiSelectSearchMode = MultiSelectSearchMode.Disabled;

...

export const enum MultiSelectSearchMode {
  Disabled = 'disabled', // Search is not available
  CaseInsensitive = 'case-insensitive', // Current available values are filtered in a case insensitive way and an emit is triggered
  EmitOnly = 'emit-only' // Current available values not filtered, but an emit still triggered
}
```

Reply:
Cool.